File: uhealthy/src/main/java/com/fmu/uhealthy/domain/Speciality.java
package com.fmu.uhealthy.domain;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import javax.persistence.*;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Entity(name="speciality")
public class Speciality {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(name = "name")
    private String name;
}

File: front/lib/model/speciality.dart
class Speciality {
  final int id;
  final String name;

  Speciality(this.id, this.name);

  static convertSpeciality(speciality) {
    return Speciality(speciality['id'], speciality['name']);
  }

  Map<String, dynamic> toJson() {
    final _data = <String, dynamic>{};
    _data['id'] = id;
    _data['name'] = name;
    return _data;
  }
}
